rclrs/src/context.rs

```diff
@@ -250,14 +273,14 @@ mod tests {
     #[test]
     fn test_create_context() -> Result<(), RclrsError> {
         // If the context fails to be created, this will cause a panic
-        let _ = Context::new(vec![])?;
+        let _ = Context::new(vec![], InitOptions::default())?;
```

Comment:
I think for building the `Context` there's enough of a difference between the args and the init options that I want to treat them differently.

In the vast majority of cases, users will want to simply pass along the CLI args from the environment, so they'll want to use [`from_env(InitOptions)`](https://github.com/ros2-rust/ros2_rust/pull/428/files#diff-d3f692f0c6fe18e3daae7e4325ea66cbafb4250b7546ccbeab0a7868e77f4484R161-R163) or [`default_from_env()`](https://github.com/ros2-rust/ros2_rust/pull/428/files#diff-d3f692f0c6fe18e3daae7e4325ea66cbafb4250b7546ccbeab0a7868e77f4484R167-R169). Trying to bundle that all up into a single InitOptions struct doesn't strike me as beneficial.

And right now the only field in `InitOptions` is `domain_id: Option<usize>` which doesn't really afford us a nice way to implicitly cast it into an `InitOptions`. For example, if a user just passes in `Some(10)` or `None` and we choose to implicitly cast that into `InitOptions`, that could be implemented just fine for now.. but it would become ambiguous and confusing if we ever add another `Option<T>` field to `InitOptions`.